coap/observer.go

```diff
@@ -36,7 +36,7 @@ type Observer struct {
 // NewObserver instantiates a new Observer.
 func NewObserver() *Observer {
 	return &Observer{
-		Messages: make(chan mainflux.Message),
+		Messages: make(chan broker.Message),
```

Comment:
maybe we `expired` and `msgID` should be in its own structure with its own mutex, and Observer can embed those structures.
Also wouldnt be better that Observer is interface?